### ISSUE TITLE
Shows wrong icon when icon is svgz

### ISSUE THREAD

**Issue by probonopd** [AppImage/libappimage#132]
[kdenlive-19.04.2d-x86_64.appimage](https://files.kde.org/kdenlive/release/kdenlive-19.04.2d-x86_64.appimage) contains `usr/share/icons/hicolor/scalable/apps/kdenlive.svgz`.

Yet just a weird placeholder icon gets shown:

![placeholder](https://user-images.githubusercontent.com/2480569/59622591-856d6c80-9121-11e9-84ae-bf0fe78e5d22.png)

Tested with

```
me@host:~$ appimaged --version
appimaged, continuous build (commit 8f26d64), build 204 built on 2019-05-31 15:04:05 UTC
```

cc @azubieta

**Comment by azubieta** [comment]
@probonopd in this case no icon was generated. The `svgz` format is not supported and it is not part of the standard see https://standards.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#file_formats

We should encourage AppImage creators to stick to the standard. IMHO this is an issue for appimagelint

**Comment by TheAssassin** [comment]
appimagelint already complains about this.

**Comment by probonopd** [comment]
Only that the Kdenlive people are not running `appimagelint`... should we make it mandatory in `appimagetool`?

**Comment by azubieta** [comment]
I guess that the whole `appimagelint` should be part of `appimagetool`. Actually I would prefer to see all the AppImage creation tools (at least the one we maintain) blended into one.

Users still complain of having too many options

**Comment by TheAssassin** [comment]
> Users still complain of having too many options

There's no evidence for such a claim.

Unix has always followed the motto "make an app so it does exactly one thing and does it right". There's no need to bloat tools with functionality.

**Comment by azubieta** [comment]
> There's no evidence for such a claim.

I guess that IRC chat logs can serve as evidence. Every day shows up a new person asking  whether  to use appimagetool, linuxdeploy or linuxdeployqt 

It's not about making feature bloated tools but to create a single tool to create AppImage. 
I guess that a similar rationale took you to create AppImageCraft, one tool to make AppImages right.

**Comment by probonopd** [comment]
I think it's good to separate concerns in separate tools, but then make the higher-level tools automatically run the lower-level ones. E.g., if `appimagetool` is the tool to make AppImages, it should bundle and invoke the tool to verify AppDirs before converting them to AppImages. So that the user doesn't have to do an extra step manually.

**Comment by TheAssassin** [comment]
> I guess that IRC chat logs can serve as evidence. Every day shows up a new person asking whether to use appimagetool, linuxdeploy or linuxdeployqt

That's purely a docs issue. If linuxdeployqt went away people wouldn't be confused by it.

> It's not about making feature bloated tools but to create a single tool to create AppImage.
> I guess that a similar rationale took you to create AppImageCraft, one tool to make AppImages right.

Not exactly, but similar. appimagecraft doesn't do anything by itself really, it just calls other tools the right way in the right order. It doesn't ship other tools or anything. It doesn't "come" with anything. You can enhance your workflow by putting the tools nearby and then enable them.

I think appimagecraft is the way to go here. It should just call appimagelint on the final build product.

**Comment by probonopd** [comment]
> if linuxdeployqt went away

That will happen only once we have a drop-in replacement for it that has a compatible CLI. E.g., a wrapper that uses `linuxdeploy` under the hood. Hundreds of projects don't want to touch their build scripts and change them to a tool that needs to be invoked totally differently.

**Comment by TheAssassin** [comment]
They only have to change their build scripts because someone provided them with some scripts they never invested 5 minutes in trying to understand that magically do stuff...

**Comment by probonopd** [comment]
Exactly. They should not need to care. It just needs to work.

**Comment by TheAssassin** [comment]
Developers must know how things work. Even if just for compliance reasons. AppImage has a tendency to create many many license compliance problems.

**Comment by probonopd** [comment]
Exactly as many as ISO or ZIP.

**Comment by TheAssassin** [comment]
Does that make it any better?

Also there's that problem of the runtime and a few other files being licensed under the terms of the MIT license, which *in theory* requires attribution inside the packaged applications. Therefore I want to write any new version of the runtime with something more permissive, e.g., the zlib license, which doesn't require attribution when used in binary form. The IP in the runtime isn't really worth requiring the MIT license. We can be more liberal there.

**Comment by TheAssassin** [comment]
Anyway, all of this is off-topic. Please focus on the actual issue again.

I think we can close this issue until there are better rules on this in the AppImage spec (e.g., when using PNG for `.DirIcon` is a rule, not a recommendation). You might also create an issue in AppImageKit about this. It's certainly not a libappimage issue, though.

**Comment by probonopd** [comment]
> MIT license, which in theory requires attribution inside the packaged applications

Please open an issue about this; if this really turns out to be a concern for anyone we can either amend the license for `runtime.c` or switch it to Public Domain entirely.

**Comment by TheAssassin** [comment]
Closing *this* issue as resolved.
